src/Surfnet/ServiceProviderDashboard/Domain/Entity/IdentityProvider.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * Copyright 2019 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Domain\Entity;
+
+use Webmozart\Assert\Assert;
```

Comment:
Please refrain from using these third party dependencies in the domain layer. The only exception we allowed during design time where the Symfony validator.

Reply:
Removed the checks

Reply:
Why not keep the checks, but make them pure PHP checks?

Reply:
I've added the checks again. I don't agree however that third party dependencies in the domain layer shouldn't be used. Because static methods are used in the constructor of the DO and `\InvalidArgumentException` exceptions are thrown this won't bubble up to other layers so this won't be any problem whatsoever. It also makes sure that the domain layer is consistent and will prevent duplicate code for validation on a lot of other places that will create a DO.

I also think that it will be a lot better to use validation in the constructor than the setters currently being used withourt any validation in the domainlayer which could lead to bugs rapidly when adding parameters.

The validators and doctrine are bubbling up from the domain layer to other layers were they break the layered architecture principle completely and therefore this would be an issue unlike the use of the assertion library.

Reply:
> I've added the checks again. I don't agree however that third party dependencies in the domain layer shouldn't be used. Because static methods are used in the constructor of the DO and \InvalidArgumentException exceptions are thrown this won't bubble up to other layers so this won't be any problem whatsoever. 

Granted, there will not be any big coupling issues here, but we try to keep coupling to an absolute minimum in the domain layer. One of the ideas of the domain layer is that is should be so loosely coupled, that you could re-use it in another application without having to install third-party dependencies. Granted we broke this rule with the Symfony validators, but that is/was a well-considered decision. 

> It also makes sure that the domain layer is consistent and will prevent duplicate code for validation on a lot of other places that will create a DO.

Yes

> I also think that it will be a lot better to use validation in the constructor than the setters currently being used withourt any validation in the domainlayer which could lead to bugs rapidly when adding parameters.

That would be preferable, but not always possible, as Symfony forms will try to use setters or public fields to set their data.

> The validators and doctrine are bubbling up from the domain layer to other layers were they break the layered architecture principle completely and therefore this would be an issue unlike the use of the assertion library.

See reply above.

Moving forward: my suggestion remains to change the validation logic to simple PHP checks as this is the current standard. If we want to change the validation logic in the domain layer to use a third party assertion library, I'm open for discussion. But we should implement it throughout the domain layer for consistency sake.

Reply:
Discussed IRL the pros and cons of the current solution and decided to keep it for now like this. We will create a story for the technical dept so we could address this later on and make the code more uniform.